src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+
+interface AutocompleteProps {
+  onSelected: (person: Person | null) => void;
+  debounceDelay?: number;
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = React.memo(
+  ({ onSelected, debounceDelay = 1300 }) => {
+    const [query, setQuery] = useState('');
+    const [appliedQuery, setAppliedQuery] = useState('');
+    const [showList, setShowList] = useState(false);
+
+    const applyQuery = useCallback(
+      debounce(setAppliedQuery, debounceDelay),
+      [],
+    );
+
+    const filtredPeople = useMemo(() => {
+      return peopleFromServer.filter(person =>
+        person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+      );
+    }, [appliedQuery]);
+
+    const handleFocus = () => {
+      setShowList(true);
+    };
+
+    const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      onSelected(null);
+      setQuery(event.target.value);
+      applyQuery(event.target.value);
+    };
+
+    const handleBlur = useCallback(
+      debounce(() => {
+        if (true) {
+          setShowList(false);
+        }
+      }, debounceDelay),
+      [],
+    );
```

Comment:
The same 
<img width="952" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/64750958/2436bd6e-eda4-43af-9daa-523fd64e68c4">